tests/test_D2_model_constraints.py

```diff
@@ -127,3 +156,113 @@ def test_prepare_constraint_minimal_renewable_share():
+                path_output_folder=user_input[PATH_OUTPUT_FOLDER],
+                move_copy=False,
+            )
+            logging.debug("Accessing script: C0_data_processing")
+            C0.all(dict_values)
+
+            logging.debug("Run parts of D0_modelling_and_optimization")
+            model, dict_model = D0.model_building.initialize(dict_values)
+            model = D0.model_building.adding_assets_to_energysystem_model(
+                dict_values, dict_model, model
+            )
+            local_energy_system = solph.Model(model)
+            logging.debug("Created oemof model based on created components and busses.")
+            return dict_values, local_energy_system, dict_model
+
+        self.dict_values, self.model, self.dict_model = run_parts()
+        self.exp_emission_limit = 1000
+        self.dict_values[CONSTRAINTS].update(
+            {MAXIMUM_EMISSIONS: {VALUE: self.exp_emission_limit}}
+        )
+        return
+
+    def test_constraint_maximum_emissions(self):
+        model = D2.constraint_maximum_emissions(
+            model=self.model, dict_values=self.dict_values
+        )
+        assert (
+            model.integral_limit_emission_factor.NoConstraint[0]
+            == self.exp_emission_limit
+        ), f"Either the maximum emission constraint has not been added or the wrong limit has been added; limit is {model.integral_limit_emission_factor.NoConstraint[0]}."
+
+    def test_add_constraints_maximum_emissions(self):
+        model = D2.add_constraints(
+            local_energy_system=self.model,
+            dict_values=self.dict_values,
+            dict_model=self.dict_model,
+        )
+        assert (
+            model.integral_limit_emission_factor.NoConstraint[0]
+            == self.exp_emission_limit
+        ), f"Either the maximum emission constraint has not been added or the wrong limit has been added; limit is {model.integral_limit_emission_factor.NoConstraint[0]}."
+
+    def test_add_constraints_maximum_emissions_None(self):
+        dict_values = self.dict_values.copy()
+        dict_values.update(
+            {
+                CONSTRAINTS: {
+                    MAXIMUM_EMISSIONS: {VALUE: None},
+                    MINIMAL_RENEWABLE_FACTOR: {
+                        VALUE: self.dict_values[CONSTRAINTS][MINIMAL_RENEWABLE_FACTOR][
+                            VALUE
+                        ]
+                    },
+                }
+            }
+        )
+        model = D2.add_constraints(
+            local_energy_system=self.model,
+            dict_values=self.dict_values,
+            dict_model=self.dict_model,
+        )
+        assert (
+            model.integral_limit_emission_factor.NoConstraint[0]
+            == self.exp_emission_limit
+        ), f"When maximum_emission is None, no emission constraint should be added to the ESM."
+
+    def test_add_constraints_minimal_renewable_share(self):
+        # todo to be added
+        pass
```

Comment:
```suggestion
   """
    def test_add_constraints_minimal_renewable_share(self):
        # todo to be added
        pass
    """
```
Otherwise it is counted as a test.